src/main/java/duke/storage/LoadLineParser.java

```diff
@@ -0,0 +1,62 @@
+package duke.storage;
+
+import duke.commons.exceptions.DukeException;
+import duke.model.*;
+import duke.model.user.Gender;
+import duke.model.user.User;
+
+public class LoadLineParser {
+
+    /**
+     * This function acts as a parser from the text file which is used to store data from the previous session.
+     * @param line the line input from the input file
+     * @param meals the structure that encapsulates the meal data for this session
+     */
+    public static void parse(MealList meals, String line) throws DukeException {
+        String[] splitLine = line.split("\\|", 4);
+        String taskType = splitLine[0];
+        String status = splitLine[1];
+        String description = splitLine[2];
+        String[] nutritionalValue = splitLine[3].split("\\|");
+        Meal newMeal;
+        if (taskType.equals("B")) {
```

Comment:
please pay attention to good software design aspects, SLAP, SRP coupling etc.
This method violates more than one design fundamentals